syft/generic/object.py

```diff
@@ -56,12 +56,22 @@ def describe(self, description: str) -> "AbstractObject":
         self.description = description
         return self
 
-    def tag(self, *_tags: str) -> "AbstractObject":
+    def tag(self, *tags: str) -> "AbstractObject":
         if self.tags is None:
             self.tags = set()
 
-        for new_tag in _tags:
-            self.tags.add(new_tag)
+        # Update the owner tag index
+        for tag in tags:
+            self.tags.add(tag)
+            if self.owner is not None:
+                if self.id not in self.owner._objects:
+                    self.owner.register_obj(self)
```

Comment:
Are we sure we want to do this under the hood?

Reply:
Well, if I tag smthg I want to search for it later, so it has to be registered right?
Registration is not 100% done properly for the moment so this is kind of a fix

Reply:
Sure! But if we raise an error when we try to tag an unregistered tensor, we'll see when the registration is broken and it'll force us to fix it

Reply:
Haha that's right - but I know it's broken and that's it's really tricky to solve and I don't want to do it in this PR^^
May I add a TODO instead, linking to a new issue?

Reply:
Fine by me if we add a comment to say that it should eventually be removed when everything is fixed (and if we don't forget to actually fix that) :)